www/js/services/shareLocalDBFile.ts

```diff
@@ -0,0 +1,116 @@
+import i18next from 'i18next';
+import { displayError, displayErrorMsg, logDebug, logWarn } from '../plugin/logger';
+
+function localDBHelpers(fileName: string, fileExtension: string = '.txt') {
+  async function localCopyFile() {
+    return new Promise<void>((resolve, reject) => {
+      let pathToFile, parentDirectory;
+      if (window['cordova'].platformId == 'android') {
+        parentDirectory = window['cordova'].file.dataDirectory.replace('files', 'databases');
+        pathToFile = parentDirectory.replace('file://', '') + fileName;
+      } else if (window['cordova'].platformId == 'ios') {
+        parentDirectory = window['cordova'].file.dataDirectory + '../';
+        pathToFile = 'LocalDatabase/' + fileName;
+      } else {
+        displayErrorMsg('Error: Unknown OS!');
+        throw new Error('Error: Unknown OS!');
+      }
+
+      window['resolveLocalFileSystemURL'](parentDirectory, (fs) => {
+        fs.filesystem.root.getFile(pathToFile, { create: false, exclusive: false }, (fileEntry) => {
+          // logDebug(`fileEntry ${fileEntry.nativeURL} is file? ${fileEntry.isFile.toString()}`);
+          logDebug(`fileEntry is: ${JSON.stringify(fileEntry, null, 2)}`);
+          window['resolveLocalFileSystemURL'](window['cordova'].file.cacheDirectory, (copyDir) => {
+            logDebug(`DirectoryEntry is: ${JSON.stringify(copyDir.filesystem.root, null, 2)}`);
+
+            fileEntry.copyTo(
+              copyDir.filesystem.root,
+              fileName + fileExtension,
+              (res) => {
+                logDebug(`Res: ${res}`);
+                resolve();
+              },
+              (rej) => {
+                logDebug(`Rej: ${JSON.stringify(rej, null, 2)}`);
```

Comment:
instead of `logDebug`, where the error gets put into the log file which the user can't email because of the error, please use `displayError` so that they can at least send us a screenshot.